File: src/main/java/com/fengsheng/Fsm.java
package com.fengsheng;

/**
 * 状态机
 */
public interface Fsm {
    /**
     * 处理函数
     *
     * @return 处理的结果 {@link ResolveResult} ，返回 {@code null} 表示停留在这个状态机
     */
    ResolveResult resolve();
}

File: src/main/java/Fsm.kt
package com.fengsheng

/**
 * 状态机
 */
interface Fsm {
    /**
     * 处理函数
     *
     * @return 处理的结果 [ResolveResult] ，返回 `null` 表示停留在这个状态机
     */
    fun resolve(): ResolveResult?
}